drivers/flash/flash_stm32g4x.c

```diff
@@ -99,10 +102,29 @@ static int erase_page(struct device *dev, unsigned int page)
 		return rc;
 	}
 
+#ifdef FLASH_OPTR_DBANK
+	if (page > 127) {
+		if (!(regs->OPTR & FLASH_OPTR_DBANK)) {
+			LOG_ERR("Page %d does not exist when DBANK=0", page);
+			return -EINVAL;
+		}
+
+		/* The pages to be erased is in bank 2*/
+		regs->CR |= FLASH_CR_BKER;
+		page = page - 128;
+		LOG_DBG("Erase page %d on bank 2", page);
+	} else {
+		LOG_DBG("Erase page %d on bank 1", page);
+	}
+
+
+	__ASSERT(page <= 127, "There are only 127 pages, but page is %d", page);
+#endif
+
 	/* Set the PER bit and select the page you wish to erase */
 	regs->CR |= FLASH_CR_PER;
 	regs->CR &= ~FLASH_CR_PNB_Msk;
-	regs->CR |= ((page % 256) << 3);
```

Comment:
@erwango  I removed the %256. Is that allowed for the other SOCs as well?

Reply:
Yes, this one is available on all G4 variants.
So this line is equivalent ?

Reply:
Are you sure this is correct ?

Reply:
ok, right